src/tests/utests/iotc_utest_mqtt_logic_layer_subscribe.c

```diff
@@ -254,6 +254,61 @@ IOTC_TT_TESTCASE(utest__match_topics__valid_data__match_topics_should_return_1,
   tt_want_int_op(match_topics(&a, &b), ==, 1);
 })
 
+IOTC_TT_TESTCASE(utest__match_topics__batch_tests, {
+  iotc_event_handle_t handle = iotc_make_empty_handle();
+
+  typedef struct {
+    char* subscription_topic;
+    const char* published_topic;
+    const uint8_t expected_result;
+  } iotc_match_topics_test_cases_t;
+
+  iotc_match_topics_test_cases_t test_cases[] = {
```

Comment:
@atigyi I really appreciate this format, are we okay to modify all topic match test cases to this format? I was just sticking with what was in place but I'd be open to refactoring

Reply:
:+1: